### PR TITLE
TechDocs: Use a flag to determine if we should use a local mkdocs or techdocs-container

### PR DESCRIPTION
**Pull request by hooloovooo**
## Hey, I just made a Pull Request!

This changes the techdocs generator to not try and detect if you have mkdocs available, but provide it as a flag instead. This will hopefully prevent any accidental issues caused by people having mkdocs installed already and just wanting to try out techdocs.

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Review by OrkoHunter — APPROVED** [review]
We'll wait until tomorrow for merge to give some time for more 👀

File: plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts
Comment (freben) [inline plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts:34]: Could this be driven from app-config.yaml instead?

Comment (hooloovooo) [reply plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts:34]: It could, but i thought it made sense to pass it in to the generator when you set it up in the backends techdocs file. Do you think the app-config is better?

Comment (pavel-kurnosov) [reply plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts:34]: I think it will be better -- case 1: in local machine you are using docker when you do development for example. and in production app-config you can override it with local setup instead of docker.

Comment (freben) [reply plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts:34]: I just like to keep all config in one place in general. I'll let you decide :) When people ask how to make use of a local runner it's kind of nice to just say to change a flag.

Comment (hooloovooo) [reply plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts:34]: The reason why i thought this would be a better way of doing it is that it kinda feels like bloating the app-config. It makes sense though. Will update PR. :)

Comment (freben) [reply plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts:34]: Another thought: is it even time to make a field `generator: "local"` instead of a specific bool? :) for ze fuuutuuuuure with more potential runners. But yeah yeah, bikeshedding at this point :)

Comment (JacobValdemar) [reply plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts:34]: I'm good eater way. I can see advantages with both solutions. If it's correct that you can just assign the bool, by using something like `process.env.MY_BOOL` in the backend techdocs file, then I think we should keep it away from the app-config, to not bloat it. Because then you can set a env variable to different things in dev and prod as @pavel-kurnosov requested.

Also - this is just a thought - maybe it would be relevant in the feature to make it more modular like the preparers. So you could register different generators depending on your needs. Both in regards to different methods for generating mkdocs (e.g. local vs docker container), but also maybe for generating different docs types, if you want to use something other than mkdocs for documentation. But that's probably a whole other discussion 🙂

Comment (pavel-kurnosov) [reply plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts:34]: I think setting the value to string instead with local/docker is a good approach -- current setup is still not the best both local and docker are not the best options. In the future, I see at least remote and event as other options -- where I can install container separately or subscribe to an event and do whatever I want :)

**Comment by hooloovooo** [comment]
Updated to look at app-config.yaml config instead of passing it in as an argument to the generator.

File: plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts
Comment (OrkoHunter) [inline plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts:73]: We can probably remove this `console.log`.

File: plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts
Comment (OrkoHunter) [inline plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts:48]: Looks like a useful utility function, that can live out of here.

File: plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts
Comment (hooloovooo) [reply plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts]: > We can probably remove this `console.log`.

D'oh!

File: plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts
Comment (hooloovooo) [reply plugins/techdocs-backend/src/techdocs/stages/generate/techdocs.ts:48]: It might be useful in another context, but i think i'd prefer it to be here for now. I think we should look at providing a stream in the router for the build job instead. That way we could allow any error to be handled in a more generic and flexible way.

**Comment by freben** [comment]
@hooloovooo Final remark - it seems that the new parameter is required; you probably want to add it to the app-settings in the create-app skeleton as well so that a newly scaffolded backstage works out of the box

**Comment by freben** [comment]
That goes in general for all app-settings stuff, it's easy to forget to put it in both places

**Comment by hooloovooo** [comment]
> @hooloovooo Final remark - it seems that the new parameter is required; you probably want to add it to the app-settings in the create-app skeleton as well so that a newly scaffolded backstage works out of the box

It's not required, but i can add it there anyway. It might be nice to see what's available.

**Review by freben — APPROVED** [review]

**Pull request merged**
